src/hotspot/share/opto/superword.cpp

```diff
@@ -2548,6 +2548,8 @@ static bool can_subword_truncate(Node* in, const Type* type) {
   // For shorts and chars, check an additional set of nodes.
   if (type == TypeInt::SHORT || type == TypeInt::CHAR) {
     switch (opc) {
+    case Op_ExtractS:
+    case Op_ExtractC:
```

Comment:
Are there any tests for these somewhere?

Reply:
I ended up encountering `ExtractS` when running `Short128VectorTests.java`, but I thought I should add `ExtractC` to the list as well. Unfortunately I don't think we don't have tests for that one yet because `char` doesn't vectorize yet.

Reply:
Does it even make sense to vectorize Extract? Is there any case where this actually succeeds? It would be worth filing a follow-up RFE if we don't want to spend the time on this now.

Reply:
I don't think there's a case where vectorizing Extract can succeed since it's already a vector op (extract scalar out of vector), so this case would only be needed to prevent assert failures.

Reply:
Whoops, I think that's my mistake! I put the nodes in the switch with the subword types since the base type was short/char, but I should have put it in the switch with the assert. It ended up not causing problems since the Extract nodes can't vectorize, but I've moved it to the assert switch in the latest commit.